include/seqan3/search/detail/policy_max_error.hpp

```diff
@@ -26,42 +27,86 @@ struct policy_max_error
 {
 protected:
     /*!\brief Returns a detail::search_param object filled by the information from the configuration.
+     *
      * \tparam configuration_t The search configuration type.
      * \tparam query_t Must model std::ranges::forward_range over the index's alphabet.
+     *
      * \param[in] cfg The configuration object.
      * \param[in] query The current query sequence.
      *
-     * \details
-     *
-     * If seqan3::max_error is set in the configuration, its value already is of type detail::search_param and
-     * can be returned as is.
-     * If seqan3::max_error_rate is set in the configuration, the error rates are converted to error counts
-     * based on the length of the query sequence.
+     * \throws std::invalid_argument
      */
     template <typename configuration_t, std::ranges::forward_range query_t>
-    auto max_error_counts(configuration_t const & cfg, [[maybe_unused]] query_t && query)
+    auto max_error_counts(configuration_t const & cfg, query_t && query)
     {
         using search_traits_t = search_traits<configuration_t>;
 
-        // retrieve error numbers / rates
-        detail::search_param max_error{0, 0, 0, 0};
+        detail::search_param errors{0, 0, 0, 0}; // total, substitution, insertion, deletion
 
-        if constexpr (search_traits_t::search_with_max_error)
-        {
-            return get<search_cfg::max_error>(cfg).value;
-        }
-        else if constexpr (search_traits_t::search_with_max_error_rate)
-        {
-            // NOTE: Casting doubles rounds towards zero (i.e. floor for positive numbers). Thus, given a rate of
-            // 10% and a read length of 101 the max number of errors is correctly casted from 10.1 errors to 10
-            auto & [total, subs, ins, del] = max_error;
-            std::tie(total, subs, ins, del) = std::apply([&query] (auto && ... args)
-            {
-                return std::tuple{(args * std::ranges::size(query))...};
-            }, get<search_cfg::max_error_rate>(cfg).value);
-        }
+        [[maybe_unused]] auto query_size = std::ranges::size(query);
+
+        if constexpr (search_traits_t::has_max_error_total)
+            errors.total = to_error_count(cfg.template value_or<search_cfg::max_error_total>(0), query_size);
+        if constexpr (search_traits_t::has_max_error_substitution)
+            errors.substitution = to_error_count(cfg.template value_or<search_cfg::max_error_substitution>(0),
+                                                 query_size);
+        if constexpr (search_traits_t::has_max_error_insertion)
+            errors.insertion = to_error_count(cfg.template value_or<search_cfg::max_error_insertion>(0), query_size);
+        if constexpr (search_traits_t::has_max_error_deletion)
+            errors.deletion = to_error_count(cfg.template value_or<search_cfg::max_error_deletion>(0), query_size);
```

Comment:
```suggestion
        errors.total = to_error_count(cfg.template value_or<search_cfg::max_error_total>(0), query_size);
        errors.substitution = to_error_count(cfg.template value_or<search_cfg::max_error_substitution>(0), query_size);
        errors.insertion = to_error_count(cfg.template value_or<search_cfg::max_error_insertion>(0), query_size);
        errors.deletion = to_error_count(cfg.template value_or<search_cfg::max_error_deletion>(0), query_size);
```

The value_or semantics give you exactly what you want. Either the config was specified, then you can use the this value or you use the alternative.